### PR TITLE
Delete repos of org when purge delete user (#27273)

### PR DESCRIPTION
**Pull request by 6543**
Fixes https://codeberg.org/forgejo/forgejo/issues/1514

Backports #27273

### CODE REVIEW

**Review by KN4CK3R — APPROVED** [review]

**Comment by 6543** [comment]
Hmm there might be more to backport if unittest fails again :/

**Review by kerwin612 — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

**Comment by lunny** [comment]
CI still failed.

**Comment by KazzmanK** [comment]
Should someone extend timeout for unit tests ? 
`
panic: test timed out after 20m0s
`

**Comment by KN4CK3R** [comment]
Unit tests should be fast. 20 Minutes is considerably too long.

**Comment by KazzmanK** [comment]
Failing commit missing
`ok  	code.gitea.io/gitea/routers/web/repo	8.126s	coverage: 5.3% of statements
`
Also 
`ok  	code.gitea.io/gitea/routers/web/repo	11.541s	coverage: 5.3% of statements
`
11 seconds instead of 3 on previous commits

Longest test is 2 minutes long
`ok  	code.gitea.io/gitea/modules/auth/password/hash	120.923s	coverage: 81.0% of statements
`

20 minutes of purge user seems to be a kind of infinite cycle or really big job.

File: services/repository/delete.go [hunk 1]
```diff
@@ -425,3 +425,30 @@ func RemoveRepositoryFromTeam(ctx context.Context, t *organization.Team, repoID
 
 	return committer.Commit()
 }
+
+// DeleteOwnerRepositoriesDirectly calls DeleteRepositoryDirectly for all repos of the given owner
+func DeleteOwnerRepositoriesDirectly(ctx context.Context, owner *user_model.User) error {
+	for {
+		repos, _, err := repo_model.GetUserRepositories(&repo_model.SearchRepoOptions{
```
Comment (lunny) [inline services/repository/delete.go]: You missed ctx here.

**Comment by silverwind** [comment]
> Longest test is 2 minutes long

This seems ripe for a refactor to speed it up.

File: services/repository/delete.go (hunk 1 above)
Comment (6543) [reply services/repository/delete.go]: 👀

**Review by lunny — APPROVED** [review]

**Pull request merged**